panels/site_sections/attractions.py

```diff
@@ -197,10 +209,24 @@ def cancel_signup(self, session, attendee_id, id):
             signup = session.query(AttractionSignup).get(id)
             if signup.attendee_id != attendee_id:
                 message = "You cannot cancel someone else's signup"
-            elif signup.checkin_time:
+            elif signup.is_checked_in:
                 message = "You cannot cancel a signup after you've checked in"
             else:
                 session.delete(signup)
                 session.commit()
         if message:
             return {'error': message}
+
+    @ajax
+    def opt_out(self, session, id, attractions_opt_out):
+        if cherrypy.request.method == 'POST':
+            attendee = session.query(Attendee).get(id)
+            attendee.attractions_opt_out = attractions_opt_out
+            session.commit()
```

Comment:
Although it wouldn't make any real difference here, I usually like returning something from Ajax methods, even if it's only an empty dict.  Not a big deal either way, but it's nice to be able to rely on an object being returned so that the client side can check for the presence of fields, etc.